onnxruntime/test/providers/nnapi/nnapi_basic_test.cc

```diff
@@ -67,6 +72,20 @@ TEST(NnapiExecutionProviderTest, ReshapeFlattenTest) {
 #endif
 }
 
+#if defined(__ANDROID__)
+// This is to verify the op_builders and op_support_checkers are consistent
+TEST(NnapiExecutionProviderTest, CreateOpBuilderAndOpSupportCheckerTest) {
+  const auto& op_builders = nnapi::GetOpBuilders();
```

Comment:
If the checkers and builders are 1:1 would it make sense to combine the them into one instance so you have an 'op handler' (or something like that) which is composed of the checking and building capabilities. 

Would halve the registration logic and help ensure things are consistent.

Reply:
The thing is OpBuilder cannot build for non-android platform, so I have to separate them, this is not ideal, I plan to have some more refactoring after 1.6 to make them more consistent

Reply:
OK. The macro to create the builder could just return an instance of IOpBuilder that threw if used couldn't it? That may confine the #ifdef on platform just to a few macros.

---
In reply to: [525842700](https://github.com/microsoft/onnxruntime/pull/5841#discussion_r525842700) [](ancestors = 525842700)